File: EmpregoCertoFront/src/app/models/candidatoVaga.ts
import { Vaga } from "./vaga";
import { Candidato } from "./candidato";

export class CandidatoVaga {
    constructor(vaga: Vaga, candidato: Candidato) {
        this.idVaga = vaga.id;
        this.nomeVaga = vaga.nome;
        this.idCandidato = candidato.id;
        this.nomeCandidato = candidato.nome;
        this.dataInscricao = new Date().toISOString();
    }

    id: number;
    idVaga: number;
    idCandidato: number;
    dataInscricao: string;
    nomeVaga: string;
    nomeCandidato: string;
}

File: EmpregoCertoAPI/EmpregoCertoAPI/Models/CandidatoVaga.cs
using System;
using System.Collections.Generic;

namespace EmpregoCertoAPI.Models
{
    public partial class CandidatoVaga
    {
        public int Id { get; set; }
        public int IdVaga { get; set; }
        public int IdCandidato { get; set; }
        public DateTime DataInscricao { get; set; }
        public string NomeVaga { get; set; }
        public string NomeCandidato { get; set; }

        public Vaga IdVagaNavigation { get; set; }
    }
}
